android/expoview/src/main/java/host/exp/exponent/utils/ExperienceActivityUtils.kt

```diff
@@ -0,0 +1,320 @@
+    }
+    ViewCompat.requestApplyInsets(decorView)
+  }
+
+  /**
+   * @return Effective style that is actually applied to the status bar.
+   */
+  @UiThread
+  private fun setStyle(style: String?, activity: Activity): String {
+    var appliedStatusBarStyle = STATUS_BAR_STYLE_LIGHT_CONTENT
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      val decorView = activity.window.decorView
+      var systemUiVisibilityFlags = decorView.systemUiVisibility
+      when (style) {
+        STATUS_BAR_STYLE_LIGHT_CONTENT -> {
+          systemUiVisibilityFlags =
+            systemUiVisibilityFlags and View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR.inv()
+          appliedStatusBarStyle = STATUS_BAR_STYLE_LIGHT_CONTENT
+        }
+        STATUS_BAR_STYLE_DARK_CONTENT -> {
+          systemUiVisibilityFlags = systemUiVisibilityFlags or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+          appliedStatusBarStyle = STATUS_BAR_STYLE_DARK_CONTENT
+        }
+        else -> {
+          systemUiVisibilityFlags = systemUiVisibilityFlags or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+          appliedStatusBarStyle = STATUS_BAR_STYLE_DARK_CONTENT
+        }
+      }
+      decorView.systemUiVisibility = systemUiVisibilityFlags
+    }
+    return appliedStatusBarStyle
+  }
+
+  @UiThread
+  private fun setHidden(hidden: Boolean, activity: Activity) {
+    if (hidden) {
+      activity.window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
+      activity.window.clearFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN)
+    } else {
+      activity.window.addFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN)
+      activity.window.clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
+    }
+  }
+
+  // endregion
+
+  fun setTaskDescription(
+    exponentManifest: ExponentManifest,
+    manifest: RawManifest,
+    activity: Activity
+  ) {
+    val iconUrl = manifest.getIconUrl()
+    val color = exponentManifest.getColorFromManifest(manifest)
+    exponentManifest.loadIconBitmap(
+      iconUrl,
+      object : BitmapListener {
+        override fun onLoadBitmap(bitmap: Bitmap?) {
+          // This if statement is only needed so the compiler doesn't show an error.
+          try {
+            activity.setTaskDescription(
+              TaskDescription(
+                if (manifest.getName() != null) manifest.getName() else "",
+                bitmap,
+                color
+              )
+            )
+          } catch (e: Throwable) {
+            EXL.e(TAG, e)
+          }
+        }
+      }
+    )
+  }
+
+  fun setNavigationBar(manifest: RawManifest, activity: Activity) {
+    val navBarOptions = manifest.getAndroidNavigationBarOptions() ?: return
+
+    // Set background color of navigation bar
+    val navBarColor = if (navBarOptions.has(ExponentManifest.MANIFEST_NAVIGATION_BAR_BACKGROUND_COLOR)) {
+      navBarOptions.optString(ExponentManifest.MANIFEST_NAVIGATION_BAR_BACKGROUND_COLOR)
+    } else null
```

Comment:
optString is annotated with `RecentlyNonNull` indicating that it returns non-null if it doesn't exist.

From the javadoc:

> Get an optional string associated with a key. It returns an empty string if there is no such key. If the value is not a string and is not null, then it is converted to a string.

I think previously it was ok for it to return an empty string (the null check did nothing but the color parse returned false), but it is clearer to make it nullable since that was the intent. Eventually I'd like to write some better extension methods for JSONObject that have better null semantics, and now that we use kotlin extension methods are easy to write.